src/ui/components/SpecialDistribution.js

```diff
@@ -1,38 +1,79 @@
 // @flow
-
 import React, {useState} from "react";
-import {div, fromInteger, fromFloatString, ZERO} from "../../ledger/grain";
-import {type IdentityId} from "../../ledger/identity";
+import {Button, Container, TextField} from "@material-ui/core";
+import {Alert} from "@material-ui/lab";
+import {makeStyles} from "@material-ui/core/styles";
+import {div, fromFloatString, lt, ZERO} from "../../ledger/grain";
 import {type Account} from "../../ledger/ledger";
 import AccountDropdown from "./AccountSelector";
 import {computeAllocation} from "../../ledger/grainAllocation";
 import * as uuid from "../../util/uuid";
 import type {TimestampMs} from "../../util/timestamp";
 import {useLedger} from "../utils/LedgerContext";
 
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "80%",
+    minWidth: "700px",
+    margin: "0 auto",
+    padding: "0 5em 5em",
+  },
+  arrowBody: {
+    color: theme.palette.text.primary,
+    flex: 1,
+    background: theme.palette.background.paper,
+    padding: "5px 20px",
+    display: "flex",
+    alignItems: "center",
+  },
+  triangle: {
+    width: 0,
+    height: 0,
+    background: theme.palette.background,
+    borderTop: "30px solid transparent",
+    borderBottom: "30px solid transparent",
+    borderLeft: `30px solid ${theme.palette.background.paper}`,
+  },
+  dropdownWrapper: {
+    border: `1px solid ${theme.palette.text.primary}`,
+    borderRadius: "50px",
+  },
+  centerRow: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  element: {flex: 1, margin: "20px"},
+  arrowInput: {display: "inline-block"},
+  headerText: {color: theme.palette.text.primary},
+}));
+
 export const SpecialDistribution = () => {
   const {ledger, updateLedger} = useLedger();
 
+  const classes = useStyles();
   const [credTimestamp, setCredTimestamp] = useState<TimestampMs>(+Date.now());
-  const [recipient, setRecipient] = useState<IdentityId | null>(null);
-  const [amount, setAmount] = useState<string>("0");
+  const [recipient, setRecipient] = useState<Account | null>(null);
+  const [amount, setAmount] = useState<string>("");
   const [memo, setMemo] = useState<string>("");
+  const [successMessage, setSuccessMessage] = useState<string>("");
 
-  const setRecipientFromAccount = (acct: Account | null) => {
-    setRecipient(acct ? acct.identity.id : null);
+  const resetState = () => {
+    setAmount("");
+    setMemo("");
+    setCredTimestamp(+Date.now());
```

Comment:
It's a feature, not a bug, that the timestamp and memo don't reset. Oftentimes when I"m using this tool I want to make a few distributions with the same TS and memo, e.g. "legacy Grain distribution for July" going out to a few accounts.

Reply:
ah, okay, that's fair I will revert.